block-producer/src/state_view/mod.rs

```diff
@@ -131,11 +134,14 @@ where
 /// 1. the candidate transaction doesn't modify the same account as an existing in-flight transaction
 /// 2. no consumed note's nullifier in candidate tx's consumed notes is already contained
 /// in `already_consumed_nullifiers`
+#[instrument(target = "miden-block-producer", skip_all, err)]
 fn ensure_in_flight_constraints(
     candidate_tx: SharedProvenTx,
     accounts_in_flight: &BTreeSet<AccountId>,
     already_consumed_nullifiers: &BTreeSet<Digest>,
 ) -> Result<(), VerifyTxError> {
+    debug!(target: COMPONENT, ?candidate_tx, ?accounts_in_flight, ?already_consumed_nullifiers);
```

Comment:
I would probably skip logging of `candidate_tx` - otherwise we'll log extra 50KB - 100KB of binary data which we've already logged previously.